### PR TITLE
Fix User Agent to Conform With the New Format

### PR DESCRIPTION
**Pull request by ajimae**
Fix user agent to conform with the new accepted format.

### CODE REVIEW

File: packages/sdk-client/src/sdk-middleware-user-agent/user-agent.ts [hunk 1]
```diff
@@ -1,25 +1,20 @@
...
-
+const { version } = require('root-require')('package.json')
 export default function createUserAgentMiddleware(
   options: UserAgentMiddlewareOptions
 ): Middleware {
+  const major = version.split('.')[0]
   const userAgent = createHttpUserAgent({
-    name: 'commercetools-js-sdk',
-    ...options,
+    name: `commercetools-sdk-javascript-v${major}/${version}`,
```
Comment (jenschude) [inline packages/sdk-client/src/sdk-middleware-user-agent/user-agent.ts:16]:
```suggestion
    name: `commercetools-sdk-javascript-v2/${version}`,
```

File: packages/sdk-client/src/sdk-middleware-user-agent/user-agent.ts [hunk 2]
```diff
@@ -1,25 +1,20 @@
...
-  libraryVersion?: string
-  contactUrl?: string
-  contactEmail?: string
-}
-
+const { version } = require('root-require')('package.json')
 export default function createUserAgentMiddleware(
   options: UserAgentMiddlewareOptions
 ): Middleware {
+  const major = version.split('.')[0]
```
Comment (jenschude) [inline packages/sdk-client/src/sdk-middleware-user-agent/user-agent.ts:14]:
```suggestion
```

File: packages/sdk-client/test/create-user-agent.test/user-agent.test.ts [hunk 3]
```diff
@@ -91,63 +91,52 @@ describe('validation', () => {
...
-      `commercetools-node-sdk Node.js/${process.version.slice(1)} (${
-        process.platform
-      }; ${process.arch}) my-awesome-library`
+      `commercetools-sdk-javascript-v${major}/${version} node.js/${process.version.slice(1)}`
     )
   })
   test('create user agent with library name and version', () => {
     const userAgent = createHttpUserAgent({
-      name: 'commercetools-node-sdk',
+      name: `commercetools-sdk-javascript-v${major}/${version}`,
```
Comment (jenschude) [inline packages/sdk-client/test/create-user-agent.test/user-agent.test.ts:106]:
```suggestion
      name: `commercetools-sdk-javascript-v2/${version}`,
```

File: packages/sdk-client/test/create-user-agent.test/user-agent.test.ts [hunk 4]
```diff
@@ -91,63 +91,52 @@ describe('validation', () => {
...
 
 describe('optional information', () => {
-  test('create user agent with only library name (missing version)', () => {
+  const { version } = require('root-require')('package.json')
+  const major = version.split('.')[0]
+  test('create user agent with the correct SDK', () => {
     const userAgent = createHttpUserAgent({
-      name: 'commercetools-node-sdk',
-      libraryName: 'my-awesome-library',
+      name: `commercetools-sdk-javascript-v${major}/${version}`
```
Comment (jenschude) [inline packages/sdk-client/test/create-user-agent.test/user-agent.test.ts:98]:
```suggestion
      name: `commercetools-sdk-javascript-v2/${version}`
```

File: packages/sdk-client/test/create-user-agent.test/user-agent.test.ts [hunk 5]
```diff
@@ -91,63 +91,52 @@ describe('validation', () => {
 })
 
 describe('optional information', () => {
-  test('create user agent with only library name (missing version)', () => {
+  const { version } = require('root-require')('package.json')
+  const major = version.split('.')[0]
```
Comment (jenschude) [inline packages/sdk-client/test/create-user-agent.test/user-agent.test.ts:95]:
```suggestion
  const { version } = require('root-require')('package.json')
```

File: packages/sdk-client/test/create-user-agent.test/user-agent.test.ts [hunk 6]
```diff
@@ -91,63 +91,52 @@ describe('validation', () => {
...
-      name: 'commercetools-node-sdk',
+      name: `commercetools-sdk-javascript-v${major}/${version}`,
       libraryName: 'my-awesome-library',
       libraryVersion: '1.0.0',
     })
     expect(userAgent).toBe(
-      `commercetools-node-sdk Node.js/${process.version.slice(1)} (${
-        process.platform
-      }; ${process.arch}) my-awesome-library/1.0.0`
+      `commercetools-sdk-javascript-v${major}/${version} node.js/${process.version.slice(1)} my-awesome-library/1.0.0`
```
Comment (jenschude) [inline packages/sdk-client/test/create-user-agent.test/user-agent.test.ts:111]:
```suggestion
      `commercetools-sdk-javascript-v2/${version} node.js/${process.version.slice(1)} my-awesome-library/1.0.0`
```

File: packages/sdk-client/test/create-user-agent.test/user-agent.test.ts [hunk 7]
```diff
@@ -91,63 +91,52 @@ describe('validation', () => {
...
     )
   })
   test('create user agent with contact url', () => {
     const userAgent = createHttpUserAgent({
-      name: 'commercetools-node-sdk',
+      name: `commercetools-sdk-javascript-v${major}/${version}`,
```
Comment (jenschude) [inline packages/sdk-client/test/create-user-agent.test/user-agent.test.ts:116]:
```suggestion
      name: `commercetools-sdk-javascript-v2/${version}`,
```

File: packages/sdk-client/test/create-user-agent.test/user-agent.test.ts [hunk 8]
```diff
@@ -91,63 +91,52 @@ describe('validation', () => {
...
       contactUrl: 'https://commercetools.com',
     })
     expect(userAgent).toBe(
-      `commercetools-node-sdk Node.js/${process.version.slice(1)} (${
-        process.platform
-      }; ${process.arch}) (+https://commercetools.com)`
+      `commercetools-sdk-javascript-v${major}/${version} node.js/${process.version.slice(1)} (+https://commercetools.com)`
```
Comment (jenschude) [inline packages/sdk-client/test/create-user-agent.test/user-agent.test.ts:120]:
```suggestion
      `commercetools-sdk-javascript-v2/${version} node.js/${process.version.slice(1)} (+https://commercetools.com)`
```

File: packages/sdk-client/test/create-user-agent.test/user-agent.test.ts [hunk 9]
```diff
@@ -91,63 +91,52 @@ describe('validation', () => {
...
     )
   })
   test('create user agent with contact email', () => {
     const userAgent = createHttpUserAgent({
-      name: 'commercetools-node-sdk',
+      name: `commercetools-sdk-javascript-v${major}/${version}`,
```
Comment (jenschude) [inline packages/sdk-client/test/create-user-agent.test/user-agent.test.ts:125]:
```suggestion
      name: `commercetools-sdk-javascript-v2/${version}`,
```

File: packages/sdk-client/test/create-user-agent.test/user-agent.test.ts [hunk 10]
```diff
@@ -91,63 +91,52 @@ describe('validation', () => {
...
       contactEmail: 'helpdesk@commercetools.com',
     })
     expect(userAgent).toBe(
-      `commercetools-node-sdk Node.js/${process.version.slice(1)} (${
-        process.platform
-      }; ${process.arch}) (+helpdesk@commercetools.com)`
+      `commercetools-sdk-javascript-v${major}/${version} node.js/${process.version.slice(1)} (+helpdesk@commercetools.com)`
```
Comment (jenschude) [inline packages/sdk-client/test/create-user-agent.test/user-agent.test.ts:129]:
```suggestion
      `commercetools-sdk-javascript-v2/${version} node.js/${process.version.slice(1)} (+helpdesk@commercetools.com)`
```

File: packages/sdk-client/test/create-user-agent.test/user-agent.test.ts [hunk 11]
```diff
@@ -91,63 +91,52 @@ describe('validation', () => {
...
     )
   })
   test('create user agent with full contact info', () => {
     const userAgent = createHttpUserAgent({
-      name: 'commercetools-node-sdk',
+      name: `commercetools-sdk-javascript-v${major}/${version}`,
```
Comment (jenschude) [inline packages/sdk-client/test/create-user-agent.test/user-agent.test.ts:134]:
```suggestion
      name: `commercetools-sdk-javascript-v2/${version}`,
```

File: packages/sdk-client/test/create-user-agent.test/user-agent.test.ts [hunk 12]
```diff
@@ -91,63 +91,52 @@ describe('validation', () => {
...
       contactUrl: 'https://commercetools.com',
       contactEmail: 'helpdesk@commercetools.com',
     })
     expect(userAgent).toBe(
-      `commercetools-node-sdk Node.js/${process.version.slice(1)} (${
-        process.platform
-      }; ${
-        process.arch
-      }) (+https://commercetools.com; +helpdesk@commercetools.com)`
+      `commercetools-sdk-javascript-v${major}/${version} node.js/${process.version.slice(1)} (+https://commercetools.com; +helpdesk@commercetools.com)`
```
Comment (jenschude) [inline packages/sdk-client/test/create-user-agent.test/user-agent.test.ts:139]:
```suggestion
      `commercetools-sdk-javascript-v2/${version} node.js/${process.version.slice(1)} (+https://commercetools.com; +helpdesk@commercetools.com)`
```

**Review by jenschude — CHANGES_REQUESTED** [review]
small changes

File: packages/sdk-client/src/sdk-middleware-user-agent/user-agent.ts [hunk 13]
```diff
@@ -1,25 +1,19 @@
...
-
+const { version } = require('root-require')('package.json')
 export default function createUserAgentMiddleware(
   options: UserAgentMiddlewareOptions
 ): Middleware {
   const userAgent = createHttpUserAgent({
-    name: 'commercetools-js-sdk',
-    ...options,
+    name: `commercetools-sdk-javascript-v2/${version}`,
+    // ...options,
```
Comment (jenschude) [inline packages/sdk-client/src/sdk-middleware-user-agent/user-agent.ts:16]: is there a reason for commenting options?

**Review by jenschude — APPROVED** [review]
small change

File: packages/sdk-client/src/sdk-middleware-user-agent/user-agent.ts (hunk 13 above)
Comment (ajimae) [reply packages/sdk-client/src/sdk-middleware-user-agent/user-agent.ts:16]: Yeah, 
The options are coming from the userAgent client middleware options
```
  libraryName: options.libraryName || '',
  libraryVersion: options.libraryVersion || '',
  contactUrl: options.contactUrl || '',
  contactEmail: options.contactEmail || '',
```
I figured that we don't need the customer to be passing in these data anymore, as the SDK will automatically generate the `libraryName` and `libraryVersion` by itsself.

Comment (jenschude) [reply packages/sdk-client/src/sdk-middleware-user-agent/user-agent.ts:16]: then i would removed the commented line

Comment (ajimae) [reply packages/sdk-client/src/sdk-middleware-user-agent/user-agent.ts:16]: Okay

**Pull request merged**